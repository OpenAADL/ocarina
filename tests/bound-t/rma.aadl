--  This AADL model illustrates how to conduct schedulability analysis
--  using Cheddar, and then code generation of periodic tasks.
--
--  Two periodic tasks run in parrallel, without interaction. Tasks
--  parameters allows RMA analysis
--
--  $Id: rma.aadl 450 2007-09-06 13:51:25Z hugues $

-----------------
-- Subprograms --
-----------------

subprogram Hello_Spg_1
properties
  source_language => Ada95;
  source_name     => "Hello.Hello_Spg_1";
end Hello_Spg_1;

subprogram Hello_Spg_2
properties
  source_language => Ada95;
  source_name     => "Hello.Hello_Spg_2";
end Hello_Spg_2;

-------------
-- Threads --
-------------

thread Task
end Task;

thread implementation Task.impl_1
calls {
  P_Spg : subprogram Hello_Spg_1;
};
properties
   Dispatch_Protocol                  => periodic;
   Period                             => 1000ms;
   Compute_Execution_time             => 0 ms .. 3 ms;
   Deadline                           => 1000 ms;
   Source_Stack_Size                  => 13952 B;
end Task.impl_1;

thread implementation Task.impl_2
calls {
  P_Spg : subprogram Hello_Spg_2;
};
properties
   Dispatch_Protocol                  => periodic;
   Period                             => 500ms;
   Compute_Execution_time             => 0 ms .. 3 ms;
   Deadline                           => 500 ms;
   Source_Stack_Size                  => 13952 B;
end Task.impl_2;

---------------
-- Processor --
---------------

processor cpurm
end cpurm;

processor implementation cpurm.impl
properties 
   Scheduling_Protocol => Posix_1003_Highest_Priority_First_Protocol;
end cpurm.impl;

---------------
-- Processes --
---------------

process node_a
end node_a;

process implementation node_a.impl
subcomponents
   Task1 : thread Task.impl_1;
   Task2 : thread Task.impl_2;
end node_a.impl;

------------
-- System --
------------

system rma 
end rma;

system implementation rma.ERC32
subcomponents
   node_a : process node_a.impl;
   cpu_rm : processor cpurm.impl;
properties
   Actual_Processor_Binding => reference cpu_rm applies to node_a;
end rma.ERC32;
