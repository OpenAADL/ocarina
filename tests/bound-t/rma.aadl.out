rma.aadl:65:04: Warning: Scheduling_Protocol is not a list while the corresponding property name at aadl_properties.aadl:416:03 is a list.
rma.aadl:65:04: Warning: The value of Scheduling_Protocol has been converted into a list.
Backends: warning : No assertion file found, create a new one
